geopandas/vectorized.pyx

```diff
@@ -1093,6 +1103,23 @@ cdef geom_type(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
     return out
 
 
+cdef cy_unary_union(np.ndarray[np.uintp_t, ndim=1, cast=True] geoms):
+    cdef GEOSContextHandle_t handle
+    cdef GEOSGeometry *collection
+    cdef GEOSGeometry *out
+    cdef size_t n = geoms.size
+
+    handle = get_geos_context_handle()
+
+    with nogil:
+        collection = GEOSGeom_createCollection_r(handle, GEOS_MULTIPOLYGON,
+                                                 <GEOSGeometry **> geoms.data,
+                                                 n)
+        out = GEOSUnaryUnion_r(handle, collection)
+
+    return geom_factory(<np.uintp_t> out)
```

Comment:
When I do destroy the collection things fail.  I agree with you that the wording there is confusing.  I'm not sure how best to handle this situation.  I'm inclined to leave it as is for now.